src/Dotnet.Status.Web/Controllers/AlertHookController.cs

```diff
@@ -0,0 +1,325 @@
+                    org,
+                    repo,
+                    issue.Number,
+                    InactiveAlertLabel,
+                    ActiveAlertLabel);
+
+                await TryRemove(() => client.Issue.Labels.RemoveFromIssue(org, repo, issue.Number, InactiveAlertLabel));
+                await TryCreate(() =>
+                    client.Issue.Labels.AddToIssue(org, repo, issue.Number, new[] {ActiveAlertLabel}));
+
+                _logger.LogInformation("Adding recurrence comment to  {org}/{repo}#{issueNumber}",
+                    org,
+                    repo,
+                    issue.Number);
+                IssueComment comment = await client.Issue.Comment.Create(org,
+                    repo,
+                    issue.Number,
+                    GenerateNewNotificationComment(notification));
+                _logger.LogInformation("Created comment {org}/{repo}#{issue}-issuecomment-{comment}",
+                    org,
+                    repo,
+                    issue.Id,
+                    comment.Id);
+            }
+        }
+
+        private string GenerateNewNotificationComment(GrafanaNotification notification)
+        {
+            var metricText = new StringBuilder();
+            foreach (GrafanaNotificationMatch match in notification.EvalMatches)
+            {
+                metricText.AppendLine($"  - *{match.Metric}* {match.Value}");
+            }
+            
+            string icon = GetIcon(notification);
+
+            return $@":{icon}: Metric state changed to *{notification.State}*
+
+> {notification.Message.Replace("\n", "\n> ")}
+
+{metricText}
+
+![Metric Graph]({notification.ImageUrl})
+
+[Go to rule]({notification.RuleUrl})".Replace("\r\n","\n");
+        }
+
+        private NewIssue GenerateNewIssue(GrafanaNotification notification)
+        {
+            var metricText = new StringBuilder();
+            foreach (GrafanaNotificationMatch match in notification.EvalMatches)
+            {
+                metricText.AppendLine($"  - *{match.Metric}* {match.Value}");
+            }
+
+            string icon = GetIcon(notification);
+
+            var issue = new NewIssue(notification.Title)
+            {
+                Body = $@":{icon}: Metric state changed to *{notification.State}*
+
+> {notification.Message.Replace("\n", "\n> ")}
+
+{metricText}
+
+![Metric Graph]({notification.ImageUrl})
+
+[Go to rule]({notification.RuleUrl})
+
+@{_githubOptions.Value.NotificationTarget}, please investigate
+
+== Automation information below, do not change ==
+alert-id: {GetUniqueIdentifier(notification)}
+".Replace("\r\n","\n")
+            };
+
+            issue.Labels.Add(ActiveAlertLabel);
+            foreach (string label in _githubOptions.Value.AlertLabels.OrEmpty())
+            {
+                issue.Labels.Add(label);
+            }
+
+            return issue;
+        }
+
+        private static string GetIcon(GrafanaNotification notification)
```

Comment:
I appreciate the extra flair of adding emoji.

Reply:
It's both fancy and super useful. Since you can just glance at the page and get an idea what's going on by looking for the colorful bits. :-)

---
In reply to: [344316993](https://github.com/dotnet/arcade-services/pull/756#discussion_r344316993) [](ancestors = 344316993)